src/py/pyodide/console.py

```diff
@@ -34,57 +41,50 @@ async def _load_packages_from_imports(*args):
         pass
 
 
-__all__ = ["repr_shorten"]
+__all__ = ["repr_shorten", "BANNER"]
 
 
-class _StdStream(io.TextIOWrapper):
-    """
-    Custom std stream to retdirect sys.stdout/stderr in _InteractiveConsole.
+def _banner():
+    """A banner similar to the one printed by the real Python interpreter."""
+    # copied from https://github.com/python/cpython/blob/799f8489d418b7f9207d333eac38214931bd7dcc/Lib/code.py#L214
+    cprt = 'Type "help", "copyright", "credits" or "license" for more information.'
+    version = platform.python_version()
+    build = f"({', '.join(platform.python_build())})"
+    return f"Python {version} {build} on WebAssembly VM\n{cprt}"
 
-    Parmeters
-    ---------
-    flush_callback
-        Function to call at each flush.
-    """
 
-    def __init__(
-        self, flush_callback: Callable[[str], None], name: Optional[str] = None
-    ):
-        # we just need to set internal buffer's name as
-        # it will automatically buble up to each buffer
-        internal_buffer = _CallbackBuffer(flush_callback, name=name)
-        buffer = io.BufferedWriter(internal_buffer)
-        super().__init__(buffer, line_buffering=True)  # type: ignore
+BANNER = _banner()
+del _banner
 
 
-class _CallbackBuffer(io.RawIOBase):
-    """
-    Internal _StdStream buffer that triggers flush callback.
+class redirect_stdin(_RedirectStream):
+    _stream = "stdin"
 
-    Parmeters
-    ---------
-    flush_callback
-        Function to call at each flush.
-    """
 
-    def __init__(
-        self, flush_callback: Callable[[str], None], name: Optional[str] = None
-    ):
-        self._flush_callback = flush_callback
+class WriteStream:
+    """A utility class so we can specify our own handlers for writes to sdout, stderr"""
+
+    def __init__(self, write_handler, name=None):
+        self.write_handler = write_handler
         self.name = name
 
-    def writable(self):
-        return True
+    def write(self, text):
+        self.write_handler(text)
+
+    def flush(self):
+        pass
+
 
-    def seekable(self):
-        return False
+class ReadStream:
+    def __init__(self, read_handler, name=None):
```

Comment:
Could you please add a docstring? Do you we want them to be public classes? Maybe private would be better?

Reply:
Yeah let's make them private.